kernel/src/main/scala/cats/kernel/Order.scala

```diff
@@ -194,16 +195,16 @@ object Order extends OrderFunctions {
    *
    * @see [[Order.whenEqual]]
    */
-  def whenEqualMonoid[A]: Monoid[Order[A]] =
-    new Monoid[Order[A]] {
+  def whenEqualMonoid[A]: Band[Order[A]] =
```

Comment:
I think we want `Band[Order[A]] with Monoid[Order[A]]` Band does not have `empty` but `BoundedSemilattice` is commutative.